lib/model.js

```diff
@@ -346,18 +330,18 @@ Model.prototype._replaceById = function(id, data_) {
   data_._id = id;
 
   // Apply getters
-  var data = data instanceof self.Document ? data_ : self.new(data_);
+  const data = data_ instanceof this.Document ? data_ : this.new(data_);
```

Comment:
Is it the expected behavior to use `data_ instanceof` instead of `data instanceof` here?

Reply:
You're right.
The `data` variable is not above this line, and var has hoisted up.
Actually, in automatic conversion by lebab, we could not recognize this hoist and it was converted, so we failed in the immediate test.
Also, the variable used in the ternary conditional expression is `data_`. For this reason, we have determined that the possibility of mistaking the variable to be compared is extremely high.

Reply:
Ok good, so it means that there was a bug and it is now fixed. :+1:

Reply:
@segayuu Is this change really correct?
Now I'm facing test failure https://github.com/hexojs/hexo/pull/3736 and it caused by this change.

Reply:
If this diff causes a problem, it works correctly by passing `this.new(data_)` regardless of whether the type of `data_` is `this.Document` (without requiring `data instanceof this.Document`) would have been.

Reply:
@segayuu 
It's sorry if I misunderstanding and I don't understand warehouse architecture.

This function name is `_replaceById`. I think it seems to replace model data. So, I think replace target is not `data_` but `this.data`.

```diff
- const data = data_ instanceof this.Document ? data_ : this.new(data_);
+ const data = this.data instanceof self.Document ? data_ : self.new(data_);
```

Reply:
`Model#data` is a plain object store. The key is id and the value is the result of `this.Document#toObject()`. If this is it can be expressed as `this.data: Record <string, object>` in TypeScript.

Reply:
@segayuu 
OK. I understood. I will try to read the current hexo source code again.

Reply:
@segayuu 
Thanks. I think warehouse has no problem. Caused by hexo source code. I created PR https://github.com/hexojs/hexo/pull/3739